source/structured_data.cc

```diff
@@ -604,6 +607,174 @@ namespace aspect
     }
 
 
+
+    template <int dim>
+    void
+    StructuredDataLookup<dim>::load_netcdf(const std::string &filename)
+    {
+#ifndef ASPECT_WITH_NETCDF
+      (void)filename;
+      AssertThrow(false, ExcMessage("Loading NetCDF files is only supported if ASPECT is configured with the NetCDF library!"));
+#else
+      TableIndices<dim> new_table_points;
+      std::vector<std::string> coordinate_column_names(dim);
+      std::vector<std::string> data_column_names;
+      std::vector<Table<dim,double>> data_tables;
+      std::vector<std::vector<double>> coordinate_values(dim);
+
+      int ncid;
+      int status;
+
+      status = nc_open(filename.c_str(), NC_NOWRITE, &ncid);
+      AssertThrow(status == NC_NOERR, ExcMessage("NetCDF Error occured!"));
+
+      int ndims, nvars, ngatts, unlimdimid;
+      status = nc_inq(ncid, &ndims, &nvars, &ngatts, &unlimdimid);
+      AssertThrow(status == NC_NOERR, ExcMessage("NetCDF Error occured!"));
+
+
+      for (int dimid=0; dimid<ndims; ++dimid)
+        {
+          char  name[NC_MAX_NAME];
```

Comment:
Ugh, this is solidly stuck in C land, isn't it? ;-)

Reply:
Fun fact: we are linking to the netcdf-c library in ASPECT, so this is in fact a C API. There used to be a netcdf-cxx library but that has been deprecated without a release since 2019. We used to have deal.II bindings to that, that we removed in 2018: https://github.com/dealii/dealii/pull/7392